src/manifold/src/csg_tree.cpp

```diff
@@ -340,14 +336,76 @@ void CsgOpNode::BatchBoolean(
  * `BatchBoolean` instead of using `Compose` for non-intersecting manifolds.
  */
 void CsgOpNode::BatchUnion() const {
-  std::vector<std::shared_ptr<const Manifold::Impl>> impls;
+  // INVARIANT: children_ is a vector of leaf nodes
+  // this kMaxUnionSize is a heuristic to avoid the pairwise disjoint check
+  // with O(n^2) complexity to take too long.
+  // If the number of children exceeded this limit, we will operate on chunks
+  // with size kMaxUnionSize.
```

Comment:
If performance becomes an issue here, this is exactly what `collider` is for. It does bounding box overlap checks in nlogn, in parallel.

Reply:
Yes, I think we talked about it in the previous PR: https://github.com/elalish/manifold/pull/131#issuecomment-1146846350

Reply:
Ah, right. Is there a reason you need to append boxes? Generally the purpose of the collider is to take your everything vector and find all the collisions at once (hence the parallelism).

Reply:
The problem is that we need to partition the set of boxes into sets of boxes such that each set contains disjoint boxes. The target is to minimize the number of boxes. Boxes that collided means that they have to be in different sets. Maybe I can think about how to do it efficiently later.